server/resp/src/main/java/org/infinispan/server/resp/commands/set/SINTERCARD.java

```diff
@@ -0,0 +1,95 @@
+package org.infinispan.server.resp.commands.set;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.ArgumentUtils;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.AggregateCompletionStage;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * {@link} https://redis.io/commands/sintercard/
+ *
+ * Returns the cardinality of the resulting intersection set.
+ *
+ * When provided with the optional LIMIT argument (default 0 means unlimited),
+ * if the intersection cardinality reaches limit partway through the
+ * computation,
+ * the algorithm exits returning LIMIT
+ *
+ * @since 15.0
+ */
+public class SINTERCARD extends RespCommand implements Resp3Command {
+   static Set<WrappedByteArray> EMPTY_SET = new HashSet<>();
+   static String LIMIT_OPT = "LIMIT";
+
+   public SINTERCARD() {
+      super(-3, 0, 0, 0);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+
+      var keysNum = ArgumentUtils.toInt(arguments.get(0));
+
+      try {
+         final int limit = processArgs(keysNum, arguments, handler);
+         var keys = arguments.subList(1, keysNum + 1);
+         AggregateCompletionStage<Void> acs = CompletionStages.aggregateCompletionStage();
+         var sets = SINTER.aggregateSets(handler, keys, acs);
+         return handler.stageToReturn(acs.freeze().thenApply((v) -> (long) SINTER.intersect(sets, limit).size()),
+               ctx,
+               Consumers.LONG_BICONSUMER);
+      } catch (Exception ex) { // Wrong args
```

Comment:
Throwing an exception for this kind of flow control can be expensive. Can we work by returning a negative number from `processArgs`?